gp-includes/things/glossary.php

```diff
@@ -79,6 +94,48 @@ public function by_set_id( $set_id ) {
 		return $this->one( "
 		    SELECT * FROM $this->table
 		    WHERE translation_set_id = %d LIMIT 1", $set_id );
+
+	}
+
+	/**
+	 * Retrieve entries and cache them
+	 *
+	 * @return array Array of Glossary_Entry.
+	 */
+	public function merge_with_glossary( GP_Glossary $merge ) {
+		$entry_map = array();
+		foreach( $this->get_entries() as $i => $entry ) {
+			$entry_map[ $entry->term ] = $i;
```

Comment:
A term can be used for different parts of speech so we have to use $term and $part_of_speech for the key.